src/main/scala/rocketchip/Configs.scala

```diff
@@ -260,11 +263,13 @@ class WithTestRAM extends Config(
         def addrMapEntries = Seq(
           AddrMapEntry("testram", MemSize(ramSize, MemAttr(AddrMapProt.RW))))
         def builder(
-            mmioPorts: HashMap[String, ClientUncachedTileLinkIO],
-            clientPorts: Seq[ClientUncachedTileLinkIO],
-            extra: Bundle, p: Parameters) {
+          mmioPorts: HashMap[String, ClientUncachedTileLinkIO],
+          clientPorts: Seq[ClientUncachedTileLinkIO],
+          interrupts: Seq[Bool],
+          extra: Bundle, p: Parameters) {
           val testram = Module(new TileLinkTestRAM(ramSize)(p))
           testram.io <> mmioPorts("testram")
+          interrupts.foreach(x => x := Bool(false))
```

Comment:
I'm happy to do that... How would that be coded and passed to the Corplex
and PLIC then?

On Friday, August 26, 2016, Howard Mao <notifications@github.com> wrote:

> In src/main/scala/rocketchip/Configs.scala
> <https://github.com/ucb-bar/rocket-chip/pull/226#discussion_r76444217>:
>
> >            val testram = Module(new TileLinkTestRAM(ramSize)(p))
> >            testram.io <> mmioPorts("testram")
> > +          interrupts.foreach(x => x := Bool(false))
>
> That wouldn't be an issue if the class itself sets the number of
> interrupts.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucb-bar/rocket-chip/pull/226/files/115e8edd8336b54c36565aaabfcaf9a0b880bd86#r76444217>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARCAJK1YtKC_JMIZsM_T6PUb3s0z16Hyks5qjw8TgaJpZM4Jtg8w>
> .
>


-- 
Megan A. Wachs
Engineer | SiFive, Inc
300 Brannan St, Suite 403
San Francisco, CA  94107
megan@sifive.com